src/simsoptpp/tracing.cpp

```diff
@@ -224,21 +409,23 @@ solve(RHS rhs, typename RHS::State y, double tmax, double dt, double dtmax, doub
                 double phi_shift = fak*2*M_PI + phi;
                 assert((phi_last <= phi_shift && phi_shift <= phi_current) || (phi_current <= phi_shift && phi_shift <= phi_last));
 
-                std::function<double(double)> rootfun = [&dense, &phi_shift, &temp, &phi_last](double t){
+                std::function<double(double)> rootfun = [&dense, &phi_shift, &temp, &phi_last, &flux](double t){
                     dense.calc_state(t, temp);
                     double diff = get_phi(temp[0], temp[1], phi_last)-phi_shift;
+                    if (flux) {
+                      diff = temp[2]-phi_shift;
+                    }
                     return diff;
                 };
-
-                auto root = toms748_solve(rootfun, tlast, tcurrent, phi_last - phi_shift, phi_current-phi_shift, roottol, rootmaxit);
+                auto root = toms748_solve(rootfun, tlast, tcurrent, phi_last - phi_shift, phi_current - phi_shift, roottol, rootmaxit);
                 double f0 = rootfun(root.first);
                 double f1 = rootfun(root.second);
                 double troot = std::abs(f0) < std::abs(f1) ? root.first : root.second;
                 dense.calc_state(troot, temp);
-                double rroot = std::sqrt(temp[0]*temp[0] + temp[1]*temp[1]);
-                double phiroot = std::atan2(temp[1], temp[0]);
-                if(phiroot<0)
-                    phiroot += 2*M_PI;
+                // double rroot = std::sqrt(temp[0]*temp[0] + temp[1]*temp[1]);
+                // double phiroot = std::atan2(temp[1], temp[0]);
+                // if(phiroot<0)
+                //     phiroot += 2*M_PI;
```

Comment:
I commented these lines out since they seemed to just be used for diagnostic printing. These lines are not applicable for the boozer tracing since the state does not map to (x,z,phi) directly.